### ISSUE TITLE
max_length support for HuggingFaceTextGenInference

### ISSUE THREAD

**Issue by sioan** [hwchase17/langchain#6851]
### System Info

Dell precision 7960
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.2 LTS

TL;DR
The error is reported in the error reproduction section.

Here's a guess at the solution:

HuggingFaceTextGenInference [docs](https://python.langchain.com/docs/modules/model_io/models/llms/integrations/huggingface_textgen_inference) and [code](https://github.com/hwchase17/langchain/blob/master/langchain/llms/huggingface_text_gen_inference.py#L77-L90) don't yet support  [huggingface's native max_length generation kwarg](https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.max_length)

I'm guessing  adding max_length below max_new_tokens in places like [here](https://github.com/hwchase17/langchain/blob/master/langchain/llms/huggingface_text_gen_inference.py#L142)
would provide the desired behavior? Ctrl-F for max_length shows other places the addition may be required

### Who can help?

@hwchase17

### Information

- [ ] The official example notebooks/scripts
- [x] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

The  code snippet below works for local models

```
pipe = pipeline("text-generation", model=hf_llm, tokenizer=tokenizer, max_new_tokens=200)
llm = HuggingFacePipeline(pipeline=pipe)

chain = RetrievalQAWithSourcesChain.from_chain_type(
    llm, chain_type="stuff", retriever=db.as_retriever()
)
chain(
    {"question": "What did the president say about Justice Breyer"},
    return_only_outputs=True,
)
```

However, when replacing the llm definition with this snippet

```
  llm = HuggingFaceTextGenInference(
      inference_server_url="http://hf-inference-server:80/",
      max_new_tokens=256,    
      top_k=10,
      top_p=0.95,
      typical_p=0.95,
      temperature=0.01,
      repetition_penalty=1.03,
  )
```
Yields this error 
```
ValidationError: Input validation error: `inputs` tokens + `max_new_tokens` must be <= 1512. Given: 2342 `inputs` 
tokens and 512 `max_new_tokens`
```
The code snippet that fails here works on it's own when used like this

`generated_text = llm("<|prompter|>What is the capital of Hungary?<|endoftext|><|assistant|>")`

### Expected behavior

Expecting a text based answer with no error.

**Comment by sioan** [comment]
Found solution on hf inference server side.

https://github.com/huggingface/text-generation-inference/blob/main/launcher/src/main.rs#L105

**Comment by rikirolly** [comment]
> RetrievalQAWithSourcesChain

I am facing the same problem, could you provide more details about this solution?
Thanks

**Comment by sioan** [comment]
> > RetrievalQAWithSourcesChain
> > @sioan
> > I am facing the same problem, could you provide more details about this solution?
> > Thanks

Oh, it appears the link I pasted above is no longer accurate.  
There's a  --max-total-tokens  when starting up the hf inference server that can be set.  Ctrl-F for max-total-tokens in the link above. 

Can you post the error message?

**Comment by rikirolly** [comment]
> > > RetrievalQAWithSourcesChain
> > > @sioan
> > > I am facing the same problem, could you provide more details about this solution?
> > > Thanks
> 
> Oh, it appears the link I pasted above is no longer accurate. There's a --max-total-tokens when starting up the hf inference server that can be set. Ctrl-F for max-total-tokens in the link above.
> 
> Can you post the error message?

I solved the same increasing the values to: --max-input-length=2048 --max-total-tokens=3072
Thanks a lot

**Comment by bitbanger** [comment]
> Found solution on hf inference server side.
> 
> https://github.com/huggingface/text-generation-inference/blob/main/launcher/src/main.rs#L105

By the way, you didn't link the version, but I'm guessing it was this commit based on the day you posted that:

https://github.com/huggingface/text-generation-inference/blob/70f485bf9f601b5450b00894f56e20b973d1c2e4/launcher/src/main.rs#L105
